1.fee_bumping/fee_bumping.md

```diff
@@ -153,25 +153,208 @@ Bitcoin blockchain, such as exchanges or custodians:
+> inheritance as described in the above Summary section.
+
+- implication 1: It is not sufficient to look at an individual transaction
+  to determine whether it can be replaced under this policy. A transaction
+  may be replaceable because one of its unconfirmed ancestors signals
+  opt-in RBF.
+- implication 2: A transaction's replaceability may change! If it is does
+  not signal opt-in replaceability, but one of its ancestors does, then it
+  is replaceable. If the ancestor then gets confirmed, the descendant
+  transaction becomes non-replaceable.
+
+> 2. The replacement transaction does not contain any new unconfirmed inputs
+> that did not previously appear in the mempool. (Unconfirmed inputs are
+> inputs spending outputs from currently unconfirmed transactions.)
+
+Rationale: this is an anti-DoS measure to stop a spender creating a replacement
+transaction which has a higher feerate than the original transaction, but
+adds new ancestor dependencies so that the package feerate for the transaction
+is lower.
+
+- implication 1: if a wallet runs out of confirmed transactions (ie all of its
+  coins are tied up in unconfirmed transactions), then it can't use the output
+  of some of those transactions as additional inputs to bump the fee on its other
+  transactions.
 
-### 5 conditions for RBF
-
-TODO
+> 3. The replacement transaction pays an absolute fee of at least the sum paid
+> by the original transactions.
+
+- implication 1: if outputs from the original transaction have been spent, then
+  the replacement transaction must exceed both the *feerate* and *total fee* of
+  the original transaction and its descendants. If the child transaction is large,
+  or there are many descendant transactions, then the total fee required to
+  replace the original transaction could be very large. This is known as 
+  *transaction pinning*.
+
+> 4. The replacement transaction must also pay for its own bandwidth at or
+> above the rate set by the node's minimum relay fee setting. For example,
+> if the minimum relay fee is 1 satoshi/byte and the replacement transaction
+> is 500 bytes total, then the replacement must pay a fee at least 500
+> satoshis higher than the sum of the originals.
+
+Rationale: this is an anti-DoS measure to prevent an adversary from repeatedly
+sending replacement transactions with miniscule feebumps and making the peer
+revalidate many versions of the same transaction.
+
+- implication 1: for transactions at low feerates, the proportional change
+  in feerate from bumping is large. A transaction with a feerate of
+  1 sat/byte needs to *double* its fee to satisfy this rule. For very
+  large transactions (eg large consolidations), then the amount of money
+  spent on fees could be high.
+
+> 5. The number of original transactions to be replaced and their descendant
+> transactions which will be evicted from the mempool must not exceed a
+> total of 100 transactions.
```

Comment:
@dgpv says: "rule 5 from bip125 seems to be another way to pin a tx - just create 100 descendants for it. And descendants can be low-fee."

Reply:
CPFP package limits should prevent this pinning, unless -limitancestorcount is increased from the current default value of 25.

You could hit this limit of 100 if you are replacing more than one tx directly, by spending their inputs in your tx, and then each of this displaced tx have descendants - replace 5 tx with 20 descendants each - you got 105 total.

But if you are aiming to RBF just one tx, its ancestor count will not be greater than limitancestorcount - 25.

Reply:
> CPFP package limits should prevent this pinning (or rather make it irrelevant for normal usecases), unless -limitancestorcount is increased from the current default value of 25.

Yes. Note that for something to be pinned, then it would be unreplaceable for the supermajority of nodes (if the transaction could be replaced in a significant number of nodes then it would find its way to the miners to get considered for the next block). Since almost all nodes are expected to run with the default `-limitdescendantcount`, then I don't think we can say a transaction can be pinned by 100 descendants.

I'll remove the implication entirely.